### PR TITLE
Add get_it logic

### PR DESCRIPTION
**Pull request by elizaveta-seleznyova-idf**

### CODE REVIEW

File: packages/presentation/lib/screen/home/home_page.dart [hunk 1]
```diff
@@ -127,7 +105,8 @@ class _HomePageState extends BlocScreenState<HomePage, HomeBloc> {
                         ),
                       ],
                     ),
-                    if (blocData.attempts >= 1)
+                    if (blocData.attempts >= 1 &&
+                        blocData.gameState == GameState.inGame)
```
Comment (elizaveta-seleznyova-idf) [inline packages/presentation/lib/screen/home/home_page.dart:109]: I'll change this logic in near future

File: packages/presentation/lib/screen/home/home_state.dart [hunk 2]
```diff
@@ -13,19 +13,22 @@ class HomeState {
...
   final GameState gameState;
 
   factory HomeState.init() => HomeState(
         attempts: 0,
         isSubmitButtonActive: false,
         gameState: GameState.inGame,
+        isAttemptsTextAvailable: false,
       );
 
   HomeState copy() => HomeState(
```
Comment (vladimir-hodaryonok-idf) [inline packages/presentation/lib/screen/home/home_state.dart]: do not seen, is this method is used anywhere?

**Review by vladimir-hodaryonok-idf — APPROVED** [review]

File: packages/presentation/lib/screen/home/home_state.dart (hunk 2 above)
Comment (elizaveta-seleznyova-idf) [reply packages/presentation/lib/screen/home/home_state.dart]: Sure! It's used for make text "Try again. You have .. attempts" available. 
[packages/presentation/lib/screen/home/home_page.dart] line 110

Comment (vladimir-hodaryonok-idf) [reply packages/presentation/lib/screen/home/home_state.dart]: I ask about copy() method =)  As I remember u used copyWith() , but copy() is not

File: lib/di/app_injector.dart [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// ignore: depend_on_referenced_packages
+import 'package:domain/di/domain_injector.dart';
+import 'package:presentation/di/presentation_injector.dart';
+
+Future<void> initAppInjector() async {
```
Comment (egor-slatvinskiy-idf) [inline lib/di/app_injector.dart]: what do we get asynchronously?

File: packages/presentation/pubspec.yaml [hunk 4]
```diff
@@ -15,6 +15,7 @@ dependencies:
 
   flutter_bloc: ^8.1.1
   equatable: ^2.0.5
+  get_it: ^7.2.0
```
Comment (egor-slatvinskiy-idf) [inline packages/presentation/pubspec.yaml]: package "get_it" should only be in the domain layer

File: pubspec.yaml [hunk 5]
```diff
@@ -14,6 +14,7 @@ dependencies:
     path: packages/presentation
 
   cupertino_icons: ^1.0.2
+  get_it: ^7.2.0
```
Comment (egor-slatvinskiy-idf) [inline pubspec.yaml]: -//-

**Review by egor-slatvinskiy-idf — APPROVED** [review]

File: packages/presentation/pubspec.yaml (hunk 4 above)
Comment (elizaveta-seleznyova-idf) [reply packages/presentation/pubspec.yaml]: ok, corrected

File: lib/di/app_injector.dart (hunk 3 above)
Comment (elizaveta-seleznyova-idf) [reply lib/di/app_injector.dart:5]: ok, corrected

File: pubspec.yaml (hunk 5 above)
Comment (elizaveta-seleznyova-idf) [reply pubspec.yaml]: ok, corrected

File: packages/presentation/lib/screen/home/home_state.dart (hunk 2 above)
Comment (elizaveta-seleznyova-idf) [reply packages/presentation/lib/screen/home/home_state.dart]: You highlight another line. Ok, I I really didn't used this method. I delete it.

File: lib/di/app_injector.dart (hunk 3 above)
Comment (elizaveta-seleznyova-idf) [reply lib/di/app_injector.dart]: removed it

File: packages/presentation/lib/screen/home/home_bloc.dart [hunk 6]
```diff
@@ -33,8 +39,13 @@ class HomeBlocImpl extends BlocImpl implements HomeBloc {
   final Random random = Random();
   final nullingAttempts = 0;
   final maxAttempts = 3;
+  final HomeDialogMapper _dialogMapper;
 
-  HomeBlocImpl(this.blocGenerateUseCase, this.blocCheckUseCase);
+  HomeBlocImpl(
+    this.blocGenerateUseCase,
+    this.blocCheckUseCase,
+    this._dialogMapper,
```
Comment (alexei-kruk-idf) [inline packages/presentation/lib/screen/home/home_bloc.dart]: youu need make private variable like _dialogMapper

File: packages/presentation/lib/screen/data/DialogParams.dart [hunk 7]
```diff
@@ -0,0 +1,11 @@
+class DialogParams {
```
Comment (alexei-kruk-idf) [inline packages/presentation/lib/screen/data/DialogParams.dart]: rename file dialog_params.dart

**Review by alexei-kruk-idf — APPROVED** [review]

File: lib/di/app_injector.dart [hunk 8]
```diff
@@ -0,0 +1,8 @@
+// ignore: depend_on_referenced_packages
```
Comment (sergei-mikhailovskii-idf) [inline lib/di/app_injector.dart]: why do u need this comment?

File: packages/presentation/lib/screen/home/home_bloc.dart [hunk 9]
```diff
@@ -129,6 +127,26 @@ class HomeBlocImpl extends BlocImpl implements HomeBloc {
...
+  @override
+  void makeAttemptsTextAvailable(GameState gameState) {
+    var attemptsTextAvailable = _state.isAttemptsTextAvailable;
+    if (_state.gameState == GameState.lose) {
+      attemptsTextAvailable = false;
+    } else if (_state.gameState == GameState.win) {
+      attemptsTextAvailable = false;
+    } else if (_state.gameState == GameState.inGame) {
+      attemptsTextAvailable = true;
+    }
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/home/home_bloc.dart]: `attemptsTextAvailable = _state.gameState == GameState.inGame`

File: packages/presentation/lib/screen/home/home_page.dart [hunk 10]
```diff
@@ -122,16 +100,19 @@ class _HomePageState extends BlocScreenState<HomePage, HomeBloc> {
...
                         ),
                       ],
                     ),
-                    if (blocData.attempts >= 1)
+                    if (blocData.isAttemptsTextAvailable == true) ...[
                       Text(
                         'Try again. You have ${3 - blocData.attempts} more attempts',
                         style: const TextStyle(fontSize: 15),
                       ),
+                    ],
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/home/home_page.dart]: why do u need `...[]`

File: packages/presentation/lib/screen/home/home_page.dart [hunk 11]
```diff
@@ -122,16 +100,19 @@ class _HomePageState extends BlocScreenState<HomePage, HomeBloc> {
...
+                                      blocData.gameState);
                                 },
                           child: const Text('Submit'),
                         ),
                       ],
                     ),
-                    if (blocData.attempts >= 1)
+                    if (blocData.isAttemptsTextAvailable == true) ...[
                       Text(
                         'Try again. You have ${3 - blocData.attempts} more attempts',
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/home/home_page.dart]: create field attemptsLeft in state. this property will have a getter which will return the value `3 - attempts`. in this case we will remove some logic from ui

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]

File: packages/presentation/lib/screen/home/home_bloc.dart (hunk 6 above)
Comment (elizaveta-seleznyova-idf) [reply packages/presentation/lib/screen/home/home_bloc.dart]: corrected

File: packages/presentation/lib/screen/data/DialogParams.dart (hunk 7 above)
Comment (elizaveta-seleznyova-idf) [reply packages/presentation/lib/screen/data/DialogParams.dart]: corrected

File: packages/presentation/lib/screen/home/home_page.dart (hunk 10 above)
Comment (elizaveta-seleznyova-idf) [reply packages/presentation/lib/screen/home/home_page.dart]: I don't need it, u r right

File: lib/di/app_injector.dart (hunk 8 above)
Comment (elizaveta-seleznyova-idf) [reply lib/di/app_injector.dart]: removed it

File: packages/presentation/lib/screen/home/home_bloc.dart (hunk 9 above)
Comment (elizaveta-seleznyova-idf) [reply packages/presentation/lib/screen/home/home_bloc.dart]: resolved

File: packages/presentation/lib/screen/home/home_page.dart (hunk 11 above)
Comment (elizaveta-seleznyova-idf) [reply packages/presentation/lib/screen/home/home_page.dart]: resolved

**Review by sergei-mikhailovskii-idf — APPROVED** [review]

**Pull request merged**